File: app/src/main/java/com/hanix/randomgame/common/utils/SnackBarUtil.java
package com.hanix.randomgame.common.utils;

import android.os.Handler;
import android.os.Looper;
import android.view.View;

import com.google.android.material.snackbar.Snackbar;
import com.hanix.randomgame.common.app.GLog;

public class SnackBarUtil {

    private static Snackbar mSnackbar;

    public static void hideSnackbar() {
        if (mSnackbar != null) {
            GLog.i("Snackbar hide true!");
            mSnackbar.dismiss();
        }
    }

    public static void showSnackbar(final View view, final String msg, final int length) {
        new Handler(Looper.getMainLooper()).post(() -> {
            hideSnackbar();
            mSnackbar = Snackbar.make(view, msg, length);
            mSnackbar.show();
        });
    }

    public static void showSnackbarClick(final View view, final String msg, final int length, String clickMsg, View.OnClickListener clickListener) {
        hideSnackbar();

        new Handler(Looper.getMainLooper()).post(() -> {
            hideSnackbar();
            if (clickListener == null) {
                mSnackbar = Snackbar.make(view, msg, length).setAction(clickMsg, (v) -> mSnackbar.dismiss());
            } else {
                mSnackbar = Snackbar.make(view, msg, length).setAction(clickMsg, clickListener);
            }
            mSnackbar.show();
        });
    }
}

File: app/src/main/java/com/hanix/randomgame/common/utils/SnackBarUtil.kt
package com.hanix.randomgame.common.utils

import android.os.Handler
import android.os.Looper
import android.view.View
import com.google.android.material.snackbar.Snackbar
import com.hanix.randomgame.common.app.GLog.i

object SnackBarUtil {
    private var mSnackbar: Snackbar? = null

    /**
     * 보여지고 있는 Snackbar 다 내리기
     */
    private fun hideSnackbar() {
        if (mSnackbar != null) {
            i("Snackbar hide true!")
            mSnackbar!!.dismiss()
        }
    }

    /**
     * 일반 Snackbar 노출
     */
    fun showSnackbar(view: View?, msg: String, length: Int) {
        if (view != null) {
            Handler(Looper.getMainLooper()).post {
                hideSnackbar()
                mSnackbar = Snackbar.make(view, msg, length)
                mSnackbar!!.show()
            }
        }
    }

    fun showSnackbarClick(
        view: View?,
        msg: String?,
        length: Int,
        clickMsg: String?,
        clickListener: View.OnClickListener?
    ) {
        hideSnackbar()
        Handler(Looper.getMainLooper()).post {
            hideSnackbar()
            mSnackbar = if (clickListener == null) {
                Snackbar.make(view!!, msg!!, length).setAction(clickMsg) { }
            } else {
                Snackbar.make(view!!, msg!!, length).setAction(clickMsg, clickListener)
            }
            mSnackbar!!.show()
        }
    }
}